src/09-recursive-functions/recursive-functions.md

```diff
@@ -24,43 +28,60 @@ object Syntax {
   def wth(x: String, xdef: Exp, body: Exp) : Exp = Ap(Fun(x,body),xdef)
 
   val sumattempt = wth("sum", Fun("n", If0("n", 0, Add("n", Ap("sum", Add("n",-1))))), Ap("sum", 10))
+```
 
-  /**
   However, sumattempt won't work and yield an unbound identifier error (why?). An alternative would be to use a variant of the
-  y combinator to support recursion properly, but today we want to talk about direct support for recursion. More specifically,
+  _y_ combinator to support recursion properly, but today we want to talk about direct support for recursion. More specifically,
   we want a language construct "letrec" that is similar to "with",  except that the bound String can be used in the expression
   the String is bound to:
-  */
 
+```scala mdoc
   case class Letrec(x: String, e: Exp, body: Exp) extends Exp
+```
+
+```scala
 }
+```
+
+```scala
 import Syntax._
+```
 
-/**
 Using letrec, our example can be expressed as follows.
-*/
 
+```scala mdoc:silent
 val sum = Letrec("sum", Fun("n", If0("n", 0, Add("n", Ap("sum", Add("n",-1))))), Ap("sum", 10))
+```
 
-/**
 Let's now consider the semantics of letrec. Consider the evaluation of ``Letrec(x,e,body)`` in an environment ``env``.
 What environment should we use to evaluate e and body, respectively? Using env for e will produce a ``ClosureV(Fun("n",..."sum"'...),env)``,
-and hence the environment when evaluating body will be ``envbody = env + (x -> ClosureV(Fun("n",..."sum"...),env))``. This is bad,
-because the ``env`` in the closure does not contain a binding for ``"sum"`` and hence the recursive invocation will fail.
+and hence the environment when evaluating body will be ``envbody = env + (x -> ClosureV(Fun("n",..."sum"...),env))``.
+
+This is bad, because the ``env`` in the closure does not contain a binding for ``"sum"`` and hence the recursive invocation will fail.
 The environment in the closure must contain a mapping for ``"sum"``. Hence envbody should look like
+
+```scala
```

Comment:
Here I would leave the Scala out entirely, since this should not end up in the Scala file.